<launch>
  <!-- Bring up the UR10 robot driver -->
  <include file="$(find ur_robot_driver)/launch/ur10_bringup.launch">
    <arg name="robot_ip" value="192.168.1.102"/>
  </include>

  <!-- Delay to ensure the robot driver is fully started before launching MoveIt! -->
  <group ns="ur10">
    <param name="robot_description" command="$(find xacro)/xacro '$(find ur_description)/urdf/ur10_robot.urdf.xacro'"/>
    
    <!-- Load controllers -->
    <rosparam file="$(find ur10_moveit_config)/config/controllers.yaml" command="load"/>

    <!-- Start controller manager and spawn controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
          args="scaled_pos_joint_traj_controller joint_state_controller"/>

    <!-- Start MoveIt! -->
    <include file="$(find bdr_ur10_moveit)/launch/demo.launch">
      <!-- <arg name="limited" value="true"/> -->  <!-- ####   DIESE ARUGMENT NUTZEN IM JOINT LIMITES IN POS,VEL,ACC zu setzen   ####-->
    </include>
  </group>
</launch>
